### PR TITLE
Issue 24400

### PR DESCRIPTION
**Pull request by ghost**
### Fixed Issues (if relevant)

1. magento/magento2#24400:Issue Found on Grid Multi-Select Column

### Manual testing scenarios (*)

1. Open any grid having some records.
2. Click on Header Checkbox to select all record on current page. https://prnt.sc/p09pr0
3. Now unselect some record, on click particular row: https://prnt.sc/p09pya
4. Again click on Header Checkbox, here is the issue.
5. Now you will see all records are unselected but header checkbox is selected. https://prnt.sc/p09qh6

Expected Result is :
https://prnt.sc/p09qof

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ghost** [comment]
@DrewML  Can you please let me know why the static test failed.

**Comment by slavvka** [comment]
@magento run all tests

File: app/code/Magento/Ui/view/base/web/js/grid/columns/multiselect.js [hunk 1]
```diff
@@ -235,6 +235,12 @@ define([
         * @returns {Multiselect} Chainable.
         */
         togglePage: function () {
+            var total = this.totalRecords(),
+                selected = this.selected().length;
+
+            if (this.isPageSelected() && selected !== total) {
+                return this.selectPage();
+            }
```
Comment (ptylek) [inline app/code/Magento/Ui/view/base/web/js/grid/columns/multiselect.js:243]: Keyword `return` at line 244 should have an empty line above it.

**Review by ptylek — CHANGES_REQUESTED** [review]
Hi @prabhatrawat-webkul! Thank you for your contribution. Changes requested.

File: app/code/Magento/Ui/view/base/web/js/grid/columns/multiselect.js [hunk 2]
```diff
@@ -311,7 +318,10 @@ define([
             var total = this.totalRecords(),
                 excluded = this.excluded().length,
                 selected = this.selected().length;
-
+                
+            if (total < selected && total > -1) {
+                total =  selected;
+            }
             if (this.excludeMode()) {
```
Comment (ptylek) [inline app/code/Magento/Ui/view/base/web/js/grid/columns/multiselect.js:325]: Keyword `if` should have an empty line above it at line 325.

**Review by ptylek — CHANGES_REQUESTED** [review]

**Comment by VladimirZaets** [comment]
@prabhatrawat-webkul, I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for collaboration

**Pull request closed without merging**